Source/player.cpp

```diff
@@ -484,9 +484,62 @@ void NewPlrAnim(int pnum, BYTE *Peq, int numFrames, int Delay, int width, int nu
 	plr[pnum]._pAnimDelay = Delay;
 	plr[pnum]._pAnimWidth = width;
 	plr[pnum]._pAnimWidth2 = (width - 64) >> 1;
-	plr[pnum]._pAnimNumSkippedFrames = numSkippedFrames;
-	plr[pnum]._pAnimGameTicksSinceSequenceStarted = processAnimationPending ? -1 : 0;
-	plr[pnum]._pAnimStopDistributingAfterFrame = stopDistributingAfterFrame;
+	plr[pnum]._pAnimGameTicksSinceSequenceStarted = 0;
+	plr[pnum]._pAnimRelevantAnimationFramesForDistributing = 0;
+	plr[pnum]._pAnimGameTickModifier = 0.0f;
+
+	if (numSkippedFrames != 0 || flags != AnimationFlags::None) {
+		int relevantAnimationFramesForDistributing = numFrames; // Animation Frames that will be adjusted for the skipped Frames/GameTicks

```

Comment:
Hey, @obligaron I am trying to understand the logic, this variable represents the remaining frames, right?

Reply:
Yes you could say this. But let me clarify more:
At this point we initialize the Animation. We must know how many Frames the Animation is in total. Cause we want to display every bit of nice content (every Frame) in it.
Normally this should be total frames (`numFrames`). But in an Player-Attack Animation this would result in some glitches.
The Sword Attack Animation for Warrior has 16 Frames.
In the 9 Frames the enemy gets hit and the hit-sound is played.
So the hit and the hit-sound happens after 56% (9/16) of the Animation is finished.

If we have a skipping of 4 frames (fastest attack modifier), this means we end up with 12 shown Frames and the hit happens at frame 5 (cause skipping is always applied before the hit animation).
The hit and the hit-sound happens now after 41% of the Animation is finished.

If we would now distribute all frames equal, the hit and the hit-sound would be out of sync.
I hoped this helps.
If you have a suggestion how the comments can be improved, please let me know. 🙂

Reply:
Thanks @obligaron! I understood the code after reading a couple times. But needed to confirm if I got it right.

It's hard to suggest better comments you did a great job.

Maybe I would renaming `relevantAnimationFramesForDistributing` to `remainingFrames`. Is hard to even suggest better variable names too.

Reply:
Naming is always hard :smiley: but really important.
I would stick with `relevantAnimationFramesForDistributing` cause my brain thinks `remainingFrames` implies some sort of progress. And at this point that's not the case.

Anyway thank you for your comment. It was really motivating. :smile: